### DISCUSSION TITLE
Duda sobre dataset a elegir y los nombres de las variables en él

### DISCUSSION THREAD

**Discussion by maguiargil** [rramosp/ai4eng.v1#238] [UDEA Students]
Cordial saludo,

Cuento con un dataset obtenido del apartado de "Competencias" en Kaggle que está separado en conjuntos de entreamiento y test, cumple los requisitos de cantidad de columnas y cantidad de filas y en su descripción se indica qué variable debe predecirse. Mi duda surge del hecho de que las variables de dicho dataset están nombradas de la siguiente manera:

var_1, var_2, var_3, var_4, ... etcétera, ..., variable_a_predecir

Me pregunto si no es inconveniente para la ejecución del proyecto que los nombres de estas variables no se refieran directamente a nombres de cosas específicas (por ejemplo edad, estaura, ... etc) y por lo tanto si no hay problema con la elección de este dataset.

[Se puede acceder a dicho dataset por este enlace.](https://www.kaggle.com/competitions/santander-customer-transaction-prediction/data)

Muchas gracias.

**Comment by Felipe-RA** [comment]
Saludos @maguiargil 

> El dataset seleccionado ha de cumplir con los siguientes requisitos:
>
> al menos 5000 instancias (filas, imágenes, etc.)
>
> al menos 30 columnas
>
> al menos el 10% de las columnas han de ser categóricas
>
>al menos ha de tener un 5% de datos faltantes en al menos el 3 columnas.
>
> **Si tu dataset no cumple con alguna de las dos últimas condiciones, LO PUEDES SIMULAR. Es decir, puedes eliminar aleatoriamente un > 5% por ciento de datos en distintas columnas, o puedes convertir datos contínuos en categóricos.**

No parece que vuestro dataset tenga datos faltantes, ni columnas categoricas. 

Por lo tanto,  tendrían que simularlos si deciden ir por este dataset.

Sobre los nombres de las variables, probablemente es porque fueron anonimizados al ser datos bancarios.  Esto les puede presentar el problema de no poder tener la intuición inicial al elegir las `features` con nombres que parezcan más o menos atractivos (ids del sistema suelen ser poco relevantes para los modelos), pero con un análisis de características (como el que ofrecen herramientas como los random forest) pueden ir buscando las más importantes.

En resumen, lo pueden elegir si tienen en cuenta las consideraciones expuestas arriba

Cualquier otra duda, aquí estamos pendientes.

**Reply by maguiargil** [reply]
Cordial saludo. Gracias por la respuesta,

Es verdad lo que señala sobre el hecho de que el conjunto de datos mencionado carece de variables categóricas, siendo este un requisito. Usted menciona que se pueden simular ¿Cómo se podría lograr eso si todas las variables no-categóricas son variables reales?

Por otra parte, me puse en la búsqueda de otro conjunto de datos y [encontré este en el que las variables categóricas son mayoría](https://www.kaggle.com/competitions/prudential-life-insurance-assessment/data) e incluso interpreto que ya pasaron por un proceso de one-hot-encoding. En cuanto a la cantidad de variables y la cantidad de registros, este cumple.

En caso de que el conjunto de datos que presenté al inicio de esta pregunta no pueda usarse (no tiene variables categóricas y sea imposible simularlas) ¿Considera que este segundo dataset pueda cumplir con los requisitos?

**Comment by maguiargil** [comment]
Cordial saludo,

Estoy a la espera de una respuesta. Agradezco la atención prestada y agradeceré la respuesta que me puedan brindar.

**Reply by rramosp** [reply]
conéctate a la clase mañana y lo conversamos

**Reply by maguiargil** [reply]
Está bien profe, muchas gracias.
